File: api/python/examples/caller.py
#encoding=utf8
from zbus import Caller, Message, SingleBroker

broker = SingleBroker(host='localhost', port=15555)

c = Caller(broker=broker, mq='MyService') 

msg = Message()
msg.set_body('hello from python')
print c.invoke(msg) 

broker.destroy()

File: api/nodejs/examples/caller.js
var zbus = require("../zbus");
var Message = zbus.Message;
var MessageClient = zbus.MessageClient;
var Caller = zbus.Caller;


var client = new MessageClient("127.0.0.1:15555");
client.connect(function(){
    var caller = new Caller(client, "MyService");
    var msg = new Message();
    msg.setBody("hello world from node.js"); 
    
    caller.invoke(msg, function(res){
        console.log(res.toString());
    }); 
});


